packages/back-end/src/models/ApiKeyModel.ts

```diff
@@ -69,26 +85,108 @@ export function generateSigningKey(prefix: string = "", bytes = 32): string {
   );
 }
 
-export async function createApiKey({
+export async function createOrganizationApiKey({
+  organizationId,
+  description,
+  role = "readonly",
+}: {
+  organizationId: string;
+  description: string;
+  role: "admin" | "readonly";
+}): Promise<ApiKeyInterface> {
+  return await createApiKey({
+    organization: organizationId,
+    secret: true,
+    encryptSDK: false,
+    description,
+    environment: "",
+    project: "",
+    role,
+  });
+}
+
+export async function createUserPersonalAccessApiKey({
+  userId,
+  organizationId,
+  description,
+  environment,
+  project,
+}: {
+  userId: string;
+  organizationId: string;
+  description: string;
+  environment: string;
+  project: string;
+}): Promise<ApiKeyInterface> {
+  return await createApiKey({
+    organization: organizationId,
+    userId,
+    secret: true,
+    environment,
+    project,
+    encryptSDK: false,
+    description,
+    role: "user",
+  });
+}
+
+/**
+ * @deprecated
+ */
+export async function createLegacySdkKey({
+  organizationId,
+  environment,
+  project,
+  encryptSDK,
+  description,
+}: {
+  organizationId: string;
+  environment: string;
+  project: string;
+  encryptSDK: boolean;
+  description: string;
+}): Promise<ApiKeyInterface> {
+  return await createApiKey({
+    organization: organizationId,
+    secret: false,
+    environment,
+    project,
+    encryptSDK,
+    description,
+  });
+}
+
+/**
+ * This lower-level function should not be exported.
+ * Use either {@link createOrganizationApiKey} with role 'readonly' | 'admin' or {@link createLegacySdkKey}
+ */
+async function createApiKey({
   environment,
   project,
   organization,
   description,
   secret,
   encryptSDK,
+  userId,
+  role,
 }: {
   environment: string;
   project: string;
   organization: string;
   description: string;
   secret: boolean;
   encryptSDK: boolean;
+  userId?: string;
+  role?: string;
 }): Promise<ApiKeyInterface> {
+  // NOTE: There's a plan to migrate SDK connection-related things to the SdkConnection collection
   if (!secret && !environment) {
     throw new Error("SDK Endpoints must have an environment set");
   }
 
-  const prefix = secret ? "secret_" : `${getShortEnvName(environment)}_`;
+  const prefix = secret
+    ? "secret_" + (role ? `${role}_` : "")
+    : `${getShortEnvName(environment)}_`;
```

Comment:
This will make user access tokens look like `secret_...`, which is identical to the existing admin secret tokens in Mongo.  Should we try to differentiate these so it's easy to tell if you're using a personal key vs an organization one?

Reply:
I pushed up a change to support this. c45c6ba26ff5cafb4e0fe824301dac8110fcb0f9

<img width="426" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/d781125a-4258-455e-b6c2-8cd58b9e4f9c">

<img width="375" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/5d335ad0-7273-4c90-a8dd-0738e4c7f2a4">